app/api/schema/activities.py

```diff
@@ -1,25 +1,22 @@
-from marshmallow_jsonapi import fields
-from marshmallow_jsonapi.flask import Schema
-
 from app.api.helpers.utilities import dasherize
+from app.models.activity import Activity
+from marshmallow_sqlalchemy import SQLAlchemyAutoSchema
 
 
-class ActivitySchema(Schema):
+class ActivitySchema(SQLAlchemyAutoSchema):
     """
     Api schema for Activity Model
     """
-
     class Meta:
         """
         Meta class for Activity Api Schema
         """
-
         type_ = 'activity'
         self_view = 'v1.activity_detail'
         self_view_kwargs = {'id': '<id>'}
         inflect = dasherize
 
-    id = fields.Str(dump_only=True)
-    actor = fields.Str(allow_none=True)
-    time = fields.DateTime(allow_none=True)
-    action = fields.Str(allow_none=True)
+        model = Activity
+        dump_only = ("id",)
+        load_instance = True
+        include_fk = True
```

Comment:
All of this needs to go in meta class, not here

Reply:
I have been trying to check the issue , as per the latest documentation of marshmallow sqlalchemy all this was put under the meta class only

Reply:
That is, indeed, what I said